iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -177,40 +187,48 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
         self.conversationController?
             .messagesChangesPublisher
             .mainSink { [unowned self] changes in
-                guard let conversationController = self.conversationController,
-                      let cid = conversationController.cid else { return }
+                guard let conversationController = self.conversationController else { return }
 
+                var isUserMessageInserted = false
                 var itemsToReconfigure: [MessageSequenceItem] = []
+
                 for change in changes {
                     switch change {
+                    case .insert(let message, _):
+                        guard message.isFromCurrentUser else { break }
+                        isUserMessageInserted = true
                     case .update(let message, _):
-                        guard !message.isDeleted else  { break }
-                        itemsToReconfigure.append(.message(cid: cid, messageID: message.id))
+                        guard !message.isDeleted else { break }
+                        itemsToReconfigure.append(.message(messageID: message.id))
                     default:
                         break
                     }
                 }
-                let conversation = conversationController.conversation
-                self.dataSource.set(messageSequence: conversation,
+
+                // Once the user sends their message, we no longer need to be in the prepare state.
+                if isUserMessageInserted {
+                    self.dataSource.shouldPrepareToSend = false
+                }
+
+                self.dataSource.set(messagesController: conversationController,
                                     itemsToReconfigure: itemsToReconfigure,
                                     showLoadMore: self.shouldShowLoadMore)
             }.store(in: &self.subscriptions)
     }
 
     func scrollToMessage(with messageId: MessageId, animateScroll: Bool, animateSelection: Bool) async {
         let task = Task {
-            guard let conversationController = self.conversationController,
-                  let cid = conversationController.cid else { return }
+            guard let conversationController = self.conversationController else { return }
 
             try? await conversationController.loadNextMessages(including: messageId)
 
             guard !Task.isCancelled else { return }
 
-            let messageItem = MessageSequenceItem.message(cid: cid, messageID: messageId)
+            let messageItem: MessageSequenceItem = .message(messageID: messageId)
```

Comment:
The message sequence controller is passed to the data source, so we don't need to pass the cid through.

Reply:
love